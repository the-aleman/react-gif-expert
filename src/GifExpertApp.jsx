import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch-Man']);
    
    const onAddCategory = (onNewCategory) => {

        if( categories.includes(onNewCategory) ) return;

        // setCategories( cat => [...categories, 'Damon Slayer'] );
        setCategories( [onNewCategory, ...categories] );
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory
                // setCategories={ setCategories }
                onNewCategory={ (value) => onAddCategory(value) }
            />

            {
                categories.map( category => (
                    <GifGrid key={ category } category={ category } />
                ))
            }
        </>
    )
}
