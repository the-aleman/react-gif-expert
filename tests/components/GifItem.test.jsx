import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";


describe('Pruebas del componente <GifItem /> ', () => {

    const title = 'Saitama';
    const url = 'https://google.com/';

    test('debe hacer match con el snapshot', () => {

        const { container } = render(<GifItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();

    });

    test('debe mostrar la imagen con el URL y el ALT indicado', () => {

        const { container } = render(<GifItem title={ title } url={ url } />);
        // expect( screen.getByRole('img').src ).toBe( url );
        
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('debe mostrar el tÃ­tulo en el componente', () => {

        const { container } = render(<GifItem title={ title } url={ url } />);
        expect( screen.getByText(title) ).toBeTruthy();

    });

});