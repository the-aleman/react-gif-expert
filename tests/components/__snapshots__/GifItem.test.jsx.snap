// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifItem />  debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://google.com/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
